src/Npgsql/NpgsqlBinaryImporter.cs

```diff
@@ -128,7 +163,7 @@ public void Write<T>([AllowNull] T value)
                     : new NpgsqlParameter<T>();
             }
 
-            Write(value, p);
+            await Write(value, p, async);
```

Comment:
This method doesn't need to be async - it can just return the task from Write on the last line.